client/disputes/index.tsx

```diff
@@ -241,58 +248,137 @@ export const DisputesList = (): JSX.Element => {
 
 	const downloadable = !! rows.length;
 
-	function onDownload() {
-		const title = __( 'Disputes', 'woocommerce-payments' );
+	const onDownload = async () => {
+		setIsDownloading( true );
+		const downloadType = totalRows > rows.length ? 'endpoint' : 'browser';
 
-		const csvColumns = [
-			{
-				...headers[ 0 ],
-				label: __( 'Dispute Id', 'woocommerce-payments' ),
-			},
-			...headers.slice( 1 ),
-		];
+		if ( 'endpoint' === downloadType ) {
+			const {
+				date_before: dateBefore,
+				date_after: dateAfter,
+				date_between: dateBetween,
+				match,
+				search,
+				type_is: typeIs,
+				type_is_not: typeIsNot,
+			} = getQuery();
 
-		const csvRows = rows.map( ( row ) => {
-			return [
-				...row.slice( 0, 3 ),
-				{
-					...row[ 3 ],
-					value: disputeStatusMapping[ row[ 3 ].value ?? '' ].message,
-				},
-				{
-					...row[ 4 ],
-					value: formatStringValue(
-						( row[ 4 ].value ?? '' ).toString()
-					),
-				},
-				...row.slice( 5, 10 ),
-				{
-					...row[ 10 ],
-					value: dateI18n(
-						'Y-m-d',
-						moment( row[ 10 ].value ).toISOString()
-					),
-				},
+			const isFiltered =
+				!! dateBefore ||
+				!! dateAfter ||
+				!! dateBetween ||
+				!! typeIs ||
+				!! typeIsNot ||
+				!! search;
+
+			const confirmThreshold = 1000;
+			const confirmMessage = sprintf(
+				__(
+					"You are about to export %d disputes. If you'd like to reduce the size of your export, you can use one or more filters. Would you like to continue?",
+					'woocommerce-payments'
+				),
+				totalRows
+			);
+
+			if (
+				isFiltered ||
+				totalRows < confirmThreshold ||
+				window.confirm( confirmMessage )
+			) {
+				try {
+					const {
+						exported_disputes: exportedDisputes,
+					} = await apiFetch( {
+						path: getDisputesCSV( {
+							dateAfter,
+							dateBefore,
+							dateBetween,
+							match,
+							search,
+							typeIs,
+							typeIsNot,
+						} ),
+						method: 'POST',
+					} );
+
+					createNotice(
+						'success',
+						__(
+							'Your export will be emailed to you.',
+							'woocommerce-payments'
+						)
+					);
+
+					wcpayTracks.recordEvent( 'wcpay_transactions_download', {
```

Comment:
Should it be `disputes` instead of `transactions`? Same goes for the fields below

Similar to l.374

Reply:
Good catch, fixed this.